homeassistant/components/unifiprotect/media_source.py

```diff
@@ -0,0 +1,862 @@
+            event_type = SimpleEventType(parts.pop(0).lower())
+        except (IndexError, ValueError) as err:
+            return _bad_identifier(item.identifier, err)
+
+        if len(parts) == 0:
+            return await self._build_events_type(
+                data, camera_id, event_type, build_children=True
+            )
+
+        try:
+            time_type = IdentifierTimeType(parts.pop(0))
+        except ValueError as err:
+            return _bad_identifier(item.identifier, err)
+
+        if len(parts) == 0:
+            return _bad_identifier(item.identifier)
+
+        # {nvr_id}:browse:all|{camera_id}:all|{event_type}:recent:{day_count}
+        if time_type == IdentifierTimeType.RECENT:
+            try:
+                days = int(parts.pop(0))
+            except (IndexError, ValueError) as err:
+                return _bad_identifier(item.identifier, err)
+
+            return await self._build_recent(
+                data, camera_id, event_type, days, build_children=True
+            )
+
+        # {nvr_id}:all|{camera_id}:all|{event_type}:range:{year}:{month}
+        # {nvr_id}:all|{camera_id}:all|{event_type}:range:{year}:{month}:all|{day}
+        try:
+            start, is_month, is_all = self._parse_range(parts)
+        except (IndexError, ValueError) as err:
+            return _bad_identifier(item.identifier, err)
+
+        if is_month:
+            return await self._build_month(
+                data, camera_id, event_type, start, build_children=True
+            )
+        return await self._build_days(
+            data, camera_id, event_type, start, build_children=True, is_all=is_all
+        )
+
+    def _parse_range(self, parts: list[str]) -> tuple[date, bool, bool]:
+        day = 1
+        is_month = True
+        is_all = True
+        year = int(parts[0])
+        month = int(parts[1])
+        if len(parts) == 3:
+            is_month = False
+            if parts[2] != "all":
+                is_all = False
+                day = int(parts[2])
+
+        start = date(year=year, month=month, day=day)
+        return start, is_month, is_all
+
+    async def _resolve_event(
+        self, data: ProtectData, event_id: str, thumbnail_only: bool = False
+    ) -> BrowseMediaSource:
+        """Resolve a specific event."""
+
+        subtype = "eventthumb" if thumbnail_only else "event"
+        try:
+            event = await data.api.get_event(event_id)
+        except NvrError as err:
+            return _bad_identifier(
+                f"{data.api.bootstrap.nvr.id}:{subtype}:{event_id}", err
+            )
+
+        if event.start is None or event.end is None:
+            raise BrowseError("Event is still ongoing")
+
+        return await self._build_event(data, event, thumbnail_only)
+
+    async def get_registry(self) -> er.EntityRegistry:
+        """Get or return Entity Registry."""
+
+        if self._registry is None:
+            self._registry = await er.async_get_registry(self.hass)
```

Comment:
Please use `er.async_get` instead. The other is legacy.

Reply:
#76870